nil dereference when ks-apiserver enables  HTTPS

<!--
You don't need to remove this comment section, it's invisible on the issues page.

## General remarks

* Attention, please fill out this issues form using English only!
* 注意！GitHub Issue 仅支持英文，中文 Issue 请在 [论坛](https://kubesphere.com.cn/forum/) 提交。
* This form is to report bugs. For general usage questions you can join our Slack channel
       [KubeSphere-users](https://join.slack.com/t/kubesphere/shared_invite/enQtNTE3MDIxNzUxNzQ0LTZkNTdkYWNiYTVkMTM5ZThhODY1MjAyZmVlYWEwZmQ3ODQ1NmM1MGVkNWEzZTRhNzk0MzM5MmY4NDc3ZWVhMjE)
-->

**Describe the Bug**
When ks-apiserver enables HTTPS, there will be nil deference.
```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x20 pc=0x381c87f]


goroutine 1 [running]:
kubesphere.io/kubesphere/cmd/ks-apiserver/app/options.(*ServerRunOptions).NewAPIServer(0xc000847170, 0xc000191500, 0xc0000af4a0, 0xc000191500, 0xc000191500)
        /Users/lihui/go/src/github.com/kubesphere/kubesphere/cmd/ks-apiserver/app/options/options.go:218 +0xcbf
kubesphere.io/kubesphere/cmd/ks-apiserver/app.Run(0xc000847170, 0xc000191500, 0x0, 0x0)
        /Users/lihui/go/src/github.com/kubesphere/kubesphere/cmd/ks-apiserver/app/server.go:95 +0xa8
kubesphere.io/kubesphere/cmd/ks-apiserver/app.NewAPIServerCommand.func1(0xc000720f00, 0xc0001dee40, 0x0, 0x4, 0x0, 0x0)
        /Users/lihui/go/src/github.com/kubesphere/kubesphere/cmd/ks-apiserver/app/server.go:60 +0xbc
github.com/spf13/cobra.(*Command).execute(0xc000720f00, 0xc00018e060, 0x4, 0x4, 0xc000720f00, 0xc00018e060)
        /Users/lihui/go/pkg/mod/github.com/spf13/cobra@v0.0.5/command.go:826 +0x472
github.com/spf13/cobra.(*Command).ExecuteC(0xc000720f00, 0x1001790, 0x0, 0x0)
        /Users/lihui/go/pkg/mod/github.com/spf13/cobra@v0.0.5/command.go:914 +0x30b
github.com/spf13/cobra.(*Command).Execute(...)
        /Users/lihui/go/pkg/mod/github.com/spf13/cobra@v0.0.5/command.go:864
main.main()
        /Users/lihui/go/src/github.com/kubesphere/kubesphere/cmd/ks-apiserver/apiserver.go:29 +0x2a

```
**Versions Used**
KubeSphere: v3.1.0

--- a/cmd/ks-apiserver/app/options/options.go
+++ b/cmd/ks-apiserver/app/options/options.go
@@ -215,7 +215,11 @@ func (s *ServerRunOptions) NewAPIServer(stopCh <-chan struct{}) (*apiserver.APIS
 		if err != nil {
 			return nil, err
 		}
-		server.TLSConfig.Certificates = []tls.Certificate{certificate}
+
+		server.TLSConfig = &tls.Config{
+			Certificates: []tls.Certificate{certificate},
+		}
+		server.Addr = fmt.Sprintf(":%d", s.GenericServerRunOptions.SecurePort)
 	}
 
 	sch := scheme.Scheme

